src/main/java/frc/team2412/robot/util/InterpolatingTreeMap.java

```diff
@@ -0,0 +1,67 @@
+package frc.team2412.robot.util;
+
+import java.util.TreeMap;
+
+public class InterpolatingTreeMap extends TreeMap<Double, ShooterDataDistancePoint> {
+    // TODO: Use JSON files in deploy folder?
+
+    public InterpolatingTreeMap() {
+        super();
+    }
+
+    public InterpolatingTreeMap(ShooterDataDistancePoint[] dataPoints) {
+        super();
+        for (ShooterDataDistancePoint dataPoint : dataPoints) {
+            put(dataPoint.getDistance(), dataPoint);
+        }
+    }
+
+    public ShooterDataDistancePoint getInterpolated(Double key) {
+        ShooterDataDistancePoint value = get(key);
+
+        // Check if we have exact value
+        if (value != null) {
+            return value;
+        }
+
+        // Get nearest keys
+        Double floor = floorKey(key);
+        Double ceiling = ceilingKey(key);
+
+        ShooterDataDistancePoint floorVal;
+        ShooterDataDistancePoint ceilingVal;
+
+        if (floor == null && ceiling == null) {
+            // If the floor and ceiling keys are not present, no keys are in the map and
+            // there is nothing to interpolate.
+            return null;
```

Comment:
It would be good to print or log a message for this scenario for troubleshooting / debugging purposes.